### PR TITLE
[ioctl] Build hdwallet export command line into new ioctl

### LINKED ISSUE #3285: [ioctl] build hdwallet export command line into new ioctl
### What would you like to be added
 - Step 1: create `hdwalletexport.go` and `hdwalletexport_test.go` files in `./ioctl/newcmd/hdwallet`. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/hdwallet/hdwalletexport.go in `hdwalletexport.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/hdwallet/hdwalletexport.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `hdwalletexport` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3285 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewHdwalletExportCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
This PR will proceed after #3419 merged.

File: ioctl/newcmd/hdwallet/hdwallet.go [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 package hdwallet
 
+import "github.com/iotexproject/iotex-core/ioctl/config"
+
 // DefaultRootDerivationPath for iotex
 // https://github.com/satoshilabs/slips/blob/master/slip-0044.md
 const DefaultRootDerivationPath = "m/44'/304'"
+
+var _hdWalletConfigFile = config.ReadConfig.Wallet + "/hdwallet"
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwallet.go:15]: don't use config.ReadConfig in newcmd.  use client.Config()

File: ioctl/newcmd/hdwallet/hdwalletexport.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			if dectxtLen <= 32 {
+				return errors.New("incorrect data")
+			}
+
+			mnemonic, hash := dectxt[:dectxtLen-32], dectxt[dectxtLen-32:]
+			if !bytes.Equal(hash, util.HashSHA256(mnemonic)) {
+				return errors.New("password error")
+			}
+
+			cmd.Println(fmt.Sprintf("Mnemonic phrase: %s"+
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletexport.go:78]: use cmd.Printf

File: ioctl/newcmd/hdwallet/hdwalletexport.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		config.Chinese: "export 导出",
+	}
+)
+
+// NewHdwalletExportCmd represents the hdwallet export command
+func NewHdwalletExportCmd(client ioctl.Client) *cobra.Command {
+	use, _ := client.SelectTranslation(_hdwalletExportCmdUses)
+	short, _ := client.SelectTranslation(_hdwalletExportCmdShorts)
+
+	cmd := &cobra.Command{
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletexport.go:40]: return &cobra.Command{

File: ioctl/newcmd/hdwallet/hdwalletexport.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+			cmd.SilenceUsage = true
+			if !fileutil.FileExists(_hdWalletConfigFile) {
+				cmd.Println("Run 'ioctl hdwallet create' to create your HDWallet first.")
+				return nil
+			}
+
+			cmd.Println("Enter password")
+			password, err := client.ReadSecret()
+			if err != nil {
+				return errors.New("failed to get password")
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletexport.go:54]: errors.Wrap

File: ioctl/newcmd/hdwallet/hdwalletexport.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+			dectxtLen := len(dectxt)
+			if dectxtLen <= 32 {
+				return errors.New("incorrect data")
+			}
+
+			mnemonic, hash := dectxt[:dectxtLen-32], dectxt[dectxtLen-32:]
+			if !bytes.Equal(hash, util.HashSHA256(mnemonic)) {
+				return errors.New("password error")
+			}
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletexport.go:76]: lines #L46 - L76  is similar with https://github.com/iotexproject/iotex-core/blob/773fb82140af8c7d4bf81e38b323e6eec3e75da7/ioctl/newcmd/hdwallet/hdwalletderive.go#L28-L51
they can be combined and move into client interface.

File: ioctl/client.go [hunk 6]
```diff
@@ -293,6 +296,36 @@ func (c *client) QueryAnalyser(reqData interface{}) (*http.Response, error) {
...
+
+	enctxt, err := os.ReadFile(hdWalletConfigFile)
+	if err != nil {
+		return nil, errors.New("failed to read config")
+	}
+
+	enckey := util.HashSHA256([]byte(password))
+	dectxt, err := util.Decrypt(enctxt, enckey)
+	if err != nil {
+		return nil, errors.New("failed to decrypt")
```
Comment (huof6829) [inline ioctl/client.go:314]: errors.Wrap

**Comment by huof6829** [comment]
@LuckyPigeon   I added test for cmd and function of client.  pls refer to it. Thx.

**Review by huof6829 — APPROVED** [review]

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**